tools/c7n_tencentcloud/c7n_tencentcloud/query.py

```diff
@@ -226,50 +227,51 @@ def resources(self, params=None):
 
         if self.resource_manager.resource_type.paging_def:
             res = self.query_helper.paged_filter(self.resource_manager.config.region,
-                                            self.resource_manager.resource_type,
-                                            params)
+                                                 self.resource_manager.resource_type,
+                                                 params)
         else:
             res = self.query_helper.filter(self.resource_manager.config.region,
-                                       self.resource_manager.resource_type,
-                                       params)
+                                           self.resource_manager.resource_type,
+                                           params)
         self.augment(res)
         return res
 
     def get_permissions(self):
         return []
 
     def augment(self, resources):
-        for r in resources:
-            r["Tags"] = self.get_resource_tag(r)
-        return resources
+        return self.get_resource_tag(resources)
 
-    def get_resource_tag(self, resource):
+    def get_resource_tag(self, resources):
         """
         Get resource tag
         All resource tags need to be obtained separately
         """
-        result = []
-        resouce_id = resource[self.resource_type.id]
-        ce6 = self.get_resource_qcs(resouce_id)
-        res = self.query_helper.get_resource_tags(self.region, ce6)
-        if res:
-            # get the resource's tags
-            for tag in res[0]["Tags"]:
-                result.append({
-                    "Key": tag["TagKey"],
-                    "Value": tag["TagValue"]
-                })
-        return result
-
-    def get_resource_qcs(self, resource_id):
+        for batch in chunks(resources, self.tag_batch_size):
+            qcs_list = self.get_resource_qcs(batch)
+            tags = self.query_helper.get_resource_tags(self.region, qcs_list)
+            for res in batch:
+                for tag in tags:
```

Comment:
this type of n * m loop can be avoided by just making a qcs to resource map and loop over the tags.